### ISSUE TITLE
replace search.from and search.to with search.range?

### ISSUE THREAD

**Issue by radare** [radare/radare2#7126]
pro/con? discuss

**Comment by Maijin** [comment]
Like what `e cmd.range=0x0,0xffff` ?

I don't care this will just break tests and existing scripts and for update the update you will need to retype the whole range.

**Comment by radare** [comment]
Its just a matter of practicity. Setting one var is sometimes faster than two, but its sometimes anoying if you wanna query one value or another. The range strings support multiple expressions to be evaluated like this

Range=0-100,400-1000

**Comment by Maijin** [comment]
k

**Comment by karliss** [comment]
For the ease of  use instead of replacing it could be made an alias using getter/setter callbacks. So that modifying search.range updates search.from and search.to.

**Comment by trufae** [comment]
search.range=1-100,300-500 cannot update from/to because it have more than 2 values. so it must be something like search.in=range and then use search.range instead of from/to.

everyone is used to the from-to workflow now. but i do understand the benefits of having such  expressions to define multiple range searchs. Anyone else willing to comment here?

**Comment by ret2libc** [comment]
IMHO we don't need it and this can be closed. I agree everyone is used to `*.from`/`*.to` and I personally don't feel the need for this.

**Comment by trufae** [comment]
I think its useful, but i think it should be implemented in @@, so my proposal here is.. instead of making more confusing eval variables. We can do something like `/ foo @@[100-200,300-6000]` which will basically update search.in/from/to for each range defined with parenthesis, brackets, dashes and commas. there's something similar in @@s:

**Comment by ret2libc** [comment]
There is also `| @{from to}          temporary set from and to for commands supporting ranges`, but that allows you to set just one range.

**Comment by trufae** [comment]
Yeah i would prefer to extend @{} to support multiple ranges instead of adding another modifier. Shouldn't be hard to implement

**Comment by ret2libc** [comment]
There is just a semantic problem, which is that in general `@?` are used for a single temporary change. We usually use `@@?` for a kind of foreach statements, so it may be a bit confusing having `@{..}` support multiple ranges (I'm thinking about something like `... @{from1 to1}{from2 to2}`)

**Comment by trufae** [comment]
Yes, this seems like a reasonable syntax to me, but i like the comma-dash syntax to define ranges of things, but this is also fine. what do u think about chaining multiple @{}@{}@{} like this?

**Comment by ret2libc** [comment]
I think `@{}@{}@{}` is not that good, because it is ambiguous. Usually `@` starts a new "temporary change command" (the `@?`).

**Comment by trufae** [comment]
Agree, so i'm fine with your proposal. i can do it for the current command parser, but im not fluent enough with treesitter to go for the newshell, do you want to implement it?

**Comment by ret2libc** [comment]
I can extend your PR with the tree-sitter changes.

**Comment by ret2libc** [comment]
Though I don't consider this high priority as it is just syntax sugar.
